yt/frontends/ramses/data_structures.py

```diff
@@ -110,11 +114,31 @@ def level_count(self):
                 lvl_count += fh._level_count
         return lvl_count
 
+    @property
+    def buffered_amr_file(self):
+        if hasattr(self, '_buffered_amr_file'):
+            return self._buffered_amr_file
+        f = IO()
+        with open(self.amr_fn, "rb") as ff:
+            f.write(ff.read())
```

Comment:
Will there ever be a situation where the AMR file is too large to keep in memory? Or generally are these pretty small?

Reply:
That's very unlikely. Typically, these files are of the order of a few `MiB`. However, this may be a concern for large hydro simulations where the sum of the amr files can be fairly large (a few `GiB`). Do you think we should remove that?

Reply:
Each of the AMR files have to fit on their own processor, along with all the other data. I just can't see how a single file could ever cause an issue with memory. This should be fine.

Reply:
@mlarichardson this is a fair point. But it may be an issue if _all_ the AMR files are stored in a single RAM (which is what happens here).

Reply:
For some numbers, Horizon AGN has ~60 MB as the largest amr file.

Reply:
If you two are fine with the memory consumption then I'm ok with it. If it's possible to cache the file in memory only when we need it and then flush the cache after we're done using it that would also work.

Reply:
Oh, yes that could become an issue if you want the whole volume, or even a large bbox.

Reply:
Wdyt @mlarichardson ? Keep it that way or remove the IO thing. I'm not even sure it has a major impact on the performance.

Reply:
It probably should be removed, just in case.

Reply:
I removed this, so this should be mergeable now (if the tests pass)